components/org.wso2.carbon.identity.organization.management.handler/src/main/java/org/wso2/carbon/identity/organization/management/handler/listener/SharedRoleMgtListener.java

```diff
@@ -536,9 +536,15 @@ private void handleOrganizationAudiencedSharedRoleDeletion(List<RoleV2> rolesLis
 
         // Get each role associated applications.
         for (String mainAppRoleId : mainAppRoleIds) {
-            List<String> associatedApplicationsIds =
-                    roleManagementService.getAssociatedApplicationByRoleId(mainAppRoleId,
-                            mainApplicationTenantDomain);
+            List<String> associatedApplicationsIds;
+            try {
+                PrivilegedCarbonContext.startTenantFlow();
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(sharedAppTenantDomain, true);
```

Comment:
```suggestion
                PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(sharedAppTenantDomain, true);
```
This should be mainApplicationTenantDomain tenant flow IMO because we are querying main app's domain data

Reply:
This is now updated to use the mainApplicationTenantDomain.